### PR TITLE
log: migrate workerutil.dbworker to new log pkg

### PR DESCRIPTION
**Pull request by jhchabran**
Part of time boxed effort to migrate all logs to the new logging library. 

## Test plan

Green CI + main dry run (green: https://buildkite.com/sourcegraph/sourcegraph/builds/169408#0182d9fb-b899-47a6-a289-26655a3c4927)

### CODE REVIEW

File: enterprise/cmd/worker/internal/permissions/bitbucket_projects.go [hunk 1]
```diff
@@ -87,10 +87,10 @@ type bitbucketProjectPermissionsHandler struct {
 
 // Handle implements the workerutil.Handler interface.
 func (h *bitbucketProjectPermissionsHandler) Handle(ctx context.Context, logger log.Logger, record workerutil.Record) (err error) {
-	logger = logger.Scoped("bitbucketProjectPermissionsHandler", "handles jobs to apply explicit permissions to all repositories of a Bitbucket Project")
+	logger = logger.Scoped("BitbucketProjectPermissionsHandler", "handles jobs to apply explicit permissions to all repositories of a Bitbucket Project")
 	defer func() {
 		if err != nil {
-			logger.Error("bitbucketProjectPermissionsHandler.Handle", log.Error(err))
+			logger.Error("BitbucketProjectPermissionsHandler.Handle", log.Error(err))
```
Comment (burmudar) [inline enterprise/cmd/worker/internal/permissions/bitbucket_projects.go:93]:
```suggestion
			logger.Error("Handle", log.Error(err))
```
It's already part of the scope

File: enterprise/internal/codemonitors/background/workers.go [hunk 2]
```diff
@@ -99,11 +99,11 @@ func newActionJobResetter(_ context.Context, s edb.CodeMonitorStore, metrics cod
 			RecordResets:        metrics.resets,
 		},
 	}
-	return dbworker.NewResetter(workerStore, options)
+	return dbworker.NewResetter(logger.Scoped("dbworker.Resetter", ""), workerStore, options)
```
Comment (burmudar) [inline enterprise/internal/codemonitors/background/workers.go:102]:
```suggestion
	return dbworker.NewResetter(logger, workerStore, options)
```
inline with the others,  the scope gets defined earlier and a scoped logger is passed here

File: internal/repos/sync_worker.go [hunk 3]
```diff
@@ -74,7 +74,7 @@ func NewSyncWorker(ctx context.Context, logger log.Logger, dbHandle basestore.Tr
 		Metrics:           newWorkerMetrics(opts.PrometheusRegisterer),
 	})
 
-	resetter := dbworker.NewResetter(store, dbworker.ResetterOptions{
+	resetter := dbworker.NewResetter(logger.Scoped("dbworker.Resetter", ""), store, dbworker.ResetterOptions{
```
Comment (burmudar) [inline internal/repos/sync_worker.go:77]:
```suggestion
	resetter := dbworker.NewResetter(logger.Scoped("repo.sync.worker.Resetter", ""), store, dbworker.ResetterOptions{
```
Based on the name below ?

File: enterprise/internal/codemonitors/background/workers.go [hunk 4]
```diff
@@ -99,11 +99,11 @@ func newActionJobResetter(_ context.Context, s edb.CodeMonitorStore, metrics cod
...
 		},
 	}
-	return dbworker.NewResetter(workerStore, options)
+	return dbworker.NewResetter(logger, workerStore, options)
 }
 
-func createDBWorkerStoreForTriggerJobs(s basestore.ShareableStore) dbworkerstore.Store {
-	return dbworkerstore.New(s.Handle(), dbworkerstore.Options{
+func createDBWorkerStoreForTriggerJobs(logger log.Logger, s basestore.ShareableStore) dbworkerstore.Store {
+	return dbworkerstore.New(logger.Scoped("dbworkerstore.Store", ""), s.Handle(), dbworkerstore.Options{
```
Comment (burmudar) [inline enterprise/internal/codemonitors/background/workers.go:106]:
```suggestion
	return dbworkerstore.New(logger.Scoped("triggerJobs.dbworker.Store", ""), s.Handle(), dbworkerstore.Options{
```

File: enterprise/internal/codemonitors/background/workers.go [hunk 5]
```diff
@@ -115,8 +115,8 @@ func createDBWorkerStoreForTriggerJobs(s basestore.ShareableStore) dbworkerstore
 	})
 }
 
-func createDBWorkerStoreForActionJobs(s edb.CodeMonitorStore) dbworkerstore.Store {
-	return dbworkerstore.New(s.Handle(), dbworkerstore.Options{
+func createDBWorkerStoreForActionJobs(logger log.Logger, s edb.CodeMonitorStore) dbworkerstore.Store {
+	return dbworkerstore.New(logger.Scoped("dbworkerstore.Store", ""), s.Handle(), dbworkerstore.Options{
```
Comment (burmudar) [inline enterprise/internal/codemonitors/background/workers.go:119]:
```suggestion
	return dbworkerstore.New(logger.Scoped("actionJobs.dbworker.Store", ""), s.Handle(), dbworkerstore.Options{
```

File: enterprise/internal/insights/background/queryrunner/worker.go [hunk 6]
```diff
@@ -131,13 +131,13 @@ func getRateLimit(defaultValue rate.Limit) func() rate.Limit {
...
 // to complete.
-func NewResetter(ctx context.Context, workerStore dbworkerstore.Store, metrics dbworker.ResetterMetrics) *dbworker.Resetter {
+func NewResetter(ctx context.Context, logger log.Logger, workerStore dbworkerstore.Store, metrics dbworker.ResetterMetrics) *dbworker.Resetter {
 	options := dbworker.ResetterOptions{
 		Name:     "insights_query_runner_worker_resetter",
 		Interval: 1 * time.Minute,
 		Metrics:  metrics,
 	}
-	return dbworker.NewResetter(workerStore, options)
+	return dbworker.NewResetter(logger.Scoped("dbworker.Resetter", ""), workerStore, options)
```
Comment (burmudar) [inline enterprise/internal/insights/background/queryrunner/worker.go:140]:
```suggestion
	return dbworker.NewResetter(logger, workerStore, options)
```
The logger here is already scoped to `queryrunner.Resetter`

File: enterprise/cmd/worker/internal/permissions/bitbucket_projects.go [hunk 7]
```diff
@@ -74,8 +74,8 @@ func (j *bitbucketProjectPermissionsJob) Routines(_ context.Context, logger log.
 	rootContext := actor.WithInternalActor(context.Background())
 
 	return []goroutine.BackgroundRoutine{
-		newBitbucketProjectPermissionsWorker(rootContext, db, ConfigInst, bbProjectMetrics),
-		newBitbucketProjectPermissionsResetter(db, ConfigInst, bbProjectMetrics),
+		newBitbucketProjectPermissionsWorker(rootContext, logger.Scoped("BitbucketProjectPermissionsResetter", ""), db, ConfigInst, bbProjectMetrics),
```
Comment (burmudar) [inline enterprise/cmd/worker/internal/permissions/bitbucket_projects.go:77]:
```suggestion
		newBitbucketProjectPermissionsWorker(rootContext, logger.Scoped("BitbucketProjectPermissionsWorker", ""), db, ConfigInst, bbProjectMetrics),
```

File: enterprise/cmd/worker/internal/permissions/bitbucket_projects.go [hunk 8]
```diff
@@ -382,13 +382,13 @@ func newBitbucketProjectPermissionsResetter(db edb.EnterpriseDB, cfg *config, me
 			RecordResets:        metrics.resets,
 		},
 	}
-	return dbworker.NewResetter(workerStore, options)
+	return dbworker.NewResetter(logger.Scoped("dbworker.Resetter", ""), workerStore, options)
```
Comment (burmudar) [inline enterprise/cmd/worker/internal/permissions/bitbucket_projects.go:385]:
```suggestion
	return dbworker.NewResetter(logger, workerStore, options)
```
The given logger is already scoped to `BitbucketProjectPermissionsResetter`

File: enterprise/cmd/worker/internal/permissions/bitbucket_projects.go [hunk 9]
```diff
@@ -382,13 +382,13 @@ func newBitbucketProjectPermissionsResetter(db edb.EnterpriseDB, cfg *config, me
...
 }
 
 // createBitbucketProjectPermissionsStore creates a store that reads and writes to the explicit_permissions_bitbucket_projects_jobs table.
 // It is used by the worker and resetter.
-func createBitbucketProjectPermissionsStore(s basestore.ShareableStore, cfg *config) dbworkerstore.Store {
-	return dbworkerstore.New(s.Handle(), dbworkerstore.Options{
+func createBitbucketProjectPermissionsStore(logger log.Logger, s basestore.ShareableStore, cfg *config) dbworkerstore.Store {
+	return dbworkerstore.New(logger.Scoped("dbworkerstore.Store", ""), s.Handle(), dbworkerstore.Options{
```
Comment (burmudar) [inline enterprise/cmd/worker/internal/permissions/bitbucket_projects.go:391]:
```suggestion
	return dbworkerstore.New(logger.Scoped("BitbucketProjectPermission.Store", ""), s.Handle(), dbworkerstore.Options{
```

File: enterprise/cmd/worker/internal/batches/janitor/resetters.go [hunk 10]
```diff
@@ -3,54 +3,56 @@ package janitor
...
-func NewReconcilerWorkerResetter(workerStore dbworkerstore.Store, metrics *metrics) *dbworker.Resetter {
+func NewReconcilerWorkerResetter(logger log.Logger, workerStore dbworkerstore.Store, metrics *metrics) *dbworker.Resetter {
 	options := dbworker.ResetterOptions{
 		Name:     "batches_reconciler_worker_resetter",
 		Interval: 1 * time.Minute,
 		Metrics:  metrics.reconcilerWorkerResetterMetrics,
 	}
 
-	resetter := dbworker.NewResetter(workerStore, options)
+	resetter := dbworker.NewResetter(logger.Scoped("dbworker.Resetter", ""), workerStore, options)
```
[the remark this answers is not in the document]
Comment (burmudar) [reply enterprise/cmd/worker/internal/batches/janitor/resetters.go:19]: @jhchabran you might've missed this one

[the remark this answers is not in the document]
Comment (jhchabran) [reply enterprise/cmd/worker/internal/batches/janitor/resetters.go:19]: Oh thanks 🙏

File: enterprise/cmd/worker/internal/batches/janitor/resetters.go [hunk 11]
```diff
@@ -3,54 +3,56 @@ package janitor
...
-func NewBulkOperationWorkerResetter(workerStore dbworkerstore.Store, metrics *metrics) *dbworker.Resetter {
+func NewBulkOperationWorkerResetter(logger log.Logger, workerStore dbworkerstore.Store, metrics *metrics) *dbworker.Resetter {
 	options := dbworker.ResetterOptions{
 		Name:     "batches_bulk_worker_resetter",
 		Interval: 1 * time.Minute,
 		Metrics:  metrics.bulkProcessorWorkerResetterMetrics,
 	}
 
-	resetter := dbworker.NewResetter(workerStore, options)
+	resetter := dbworker.NewResetter(logger.Scoped("dbworker.Resetter", ""), workerStore, options)
```
Comment (burmudar) [inline enterprise/cmd/worker/internal/batches/janitor/resetters.go:32]:
```suggestion
	resetter := dbworker.NewResetter(logger, workerStore, options)
```

File: enterprise/cmd/worker/internal/batches/janitor/resetters.go [hunk 12]
```diff
@@ -3,54 +3,56 @@ package janitor
...
-func NewBatchSpecWorkspaceExecutionWorkerResetter(workerStore dbworkerstore.Store, metrics *metrics) *dbworker.Resetter {
+func NewBatchSpecWorkspaceExecutionWorkerResetter(logger log.Logger, workerStore dbworkerstore.Store, metrics *metrics) *dbworker.Resetter {
 	options := dbworker.ResetterOptions{
 		Name:     "batch_spec_workspace_execution_worker_resetter",
 		Interval: 1 * time.Minute,
 		Metrics:  metrics.batchSpecWorkspaceExecutionWorkerResetterMetrics,
 	}
 
-	resetter := dbworker.NewResetter(workerStore, options)
+	resetter := dbworker.NewResetter(logger.Scoped("dbworker.Resetter", ""), workerStore, options)
```
Comment (burmudar) [inline enterprise/cmd/worker/internal/batches/janitor/resetters.go:45]:
```suggestion
	resetter := dbworker.NewResetter(logger, workerStore, options)
```

File: enterprise/cmd/worker/internal/batches/janitor/resetters.go [hunk 13]
```diff
@@ -3,54 +3,56 @@ package janitor
...
-func NewBatchSpecWorkspaceResolutionWorkerResetter(workerStore dbworkerstore.Store, metrics *metrics) *dbworker.Resetter {
+func NewBatchSpecWorkspaceResolutionWorkerResetter(logger log.Logger, workerStore dbworkerstore.Store, metrics *metrics) *dbworker.Resetter {
 	options := dbworker.ResetterOptions{
 		Name:     "batch_changes_batch_spec_resolution_worker_resetter",
 		Interval: 1 * time.Minute,
 		Metrics:  metrics.batchSpecResolutionWorkerResetterMetrics,
 	}
 
-	resetter := dbworker.NewResetter(workerStore, options)
+	resetter := dbworker.NewResetter(logger.Scoped("dbworker.Resetter", ""), workerStore, options)
```
Comment (burmudar) [inline enterprise/cmd/worker/internal/batches/janitor/resetters.go:56]:
```suggestion
	resetter := dbworker.NewResetter(logger, workerStore, options)
```

File: enterprise/cmd/worker/internal/batches/janitor/resetters.go (hunk 11 above)
Comment (burmudar) [reply enterprise/cmd/worker/internal/batches/janitor/resetters.go:32]: for some reason in this file I forgot I can make suggestions 🤦🏼‍♂️

**Review by burmudar — APPROVED** [review]
LGTM 🎉

**Pull request merged**